### DISCUSSION TITLE
Source controller does not refresh helmcharts artifacts with rolling tags

### DISCUSSION THREAD

**Discussion by Kalimero66** [fluxcd/flux2#4260] [Q&A] [answered]
Hello,
For our dev environment, we use rolling tags for helm charts and we noticed that flux does not refresh helmchart artifacts with the latest content. The helmRepo index.yaml is updated properly though, the digest of a chart changes when its rolling tag points to new content. 
However the helmChart object never changes and get stuck in the 1st version of the rolling tag.
I even checked inside the source controller pod the various files in the filesystem 
Is there a way to force the source controller to refresh helmChart artifacts against the latest version of the chart indicated by the helmRepo index.yaml ? 
TIA.
K.

**Comment by hiddeco** [comment]
Helm charts follow Semantic Versioning, and should be immutable because of this. The proper way to manage this would be to attach [build metadata](https://semver.org/#spec-item-10), which is [supported by Helm](https://helm.sh/docs/chart_best_practices/conventions/#version-numbers).

**Reply by Kalimero66** [reply]
I perfectly understand your point, rolling tags are something we only do in dev.
So I also understand that flux has no backdoor to workaround this issue ?
I was thinking of tuning some cache TTL settings to force more frequent updates of helmChart artifacts

**Reply by stefanprodan** [reply]
You can use a GitRepository in dev, as the source for the chart. Flux will bump the chart version for new commits, see https://fluxcd.io/flux/components/source/helmcharts/#reconcile-strategy

**Reply by hiddeco** [reply, marked as answer]
It has not, because the helm-controller itself looks at the version as presented in the chart versus the version presented in the metadata of the release object. This means that even if you would be able to get a newer version of your chart, the helm-controller would still not think it would require an update.

Given the state of Helm packaging (https://github.com/helm/helm/issues/3612), taking chart digests into account is generally an issue. As a rebuild from a different environment (but with the exact same contents) could result in a false positive.

**Reply by Kalimero66** [reply]
The GitRepository option is worth looking at even though with 50+ charts it might add some overhead compared to the build metadata option.
 Thank you both @stefanprodan and @hiddeco for your quick input.
